### PR TITLE
Add WildFly Swarm recipe based on CentOS

### PR DESCRIPTION
**Pull request by dharmit**
Signed-off-by: Dharmit Shah <dshah@redhat.com>

### CODE REVIEW

File: recipes/centos_wildfly_swarm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ENV JAVA_HOME=/usr/lib/jvm/java-${JAVA_VERSION} \
+    M2_HOME=/opt/rh/rh-maven33/root/usr/share/maven
+
+RUN yum -y update && \
+    yum -y install wget sudo openssh-server centos-release-scl && \
+    yum -y install rh-maven33 java-${JAVA_VERSION}-openjdk-devel && \
+    yum clean all && \
+    sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd && \
+    echo "user ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
+    sed -i 's/Defaults    requiretty/#Defaults    requiretty/g' /etc/sudoers && \
```
Comment (eivantsov) [inline recipes/centos_wildfly_swarm/Dockerfile]:
```
-    sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd && \
+    sed -ri 's/UsePAM yes/#UsePAM yes/g' /etc/ssh/sshd_config && \
+    sed -ri 's/#UsePAM no/UsePAM no/g' /etc/ssh/sshd_config && \
```
Can you please replace it this way? I have just discovered that without these lines sshing into a container using key is impossible.

Comment (dharmit) [inline recipes/centos_wildfly_swarm/Dockerfile]: @eivantsov Sure, I can make the change. Is this about ssh from the host system on which workspace container is running? Should one be able to login to the workspace container using ssh key of the user running the container (mostly root user)? And should one be able to login as `user`? Something like below from the host system:

```bash
$ docker ps
CONTAINER ID        IMAGE                                                                           COMMAND                  CREATED             STATUS              PORTS                                                                                                                       NAMES
009e96217179        eclipse-che/workspacemzk5z182ww40cfpb_machinequ76u97ec6i5hii0_che_dev-machine   "/bin/sh -c 'tail -f "   8 hours ago         Up 8 hours          0.0.0.0:32927->22/tcp, 0.0.0.0:32926->4401/tcp, 0.0.0.0:32925->4403/tcp, 0.0.0.0:32924->4411/tcp, 0.0.0.0:32923->8080/tcp   workspacemzk5z182ww40cfpb_machinequ76u97ec6i5hii0_che_dev-machine
2c4a1e7c306b        eclipse-che/workspaceuejf119p8lml5zt4_machinerjvjqwat0s9rid3r_che_dev-machine   "/bin/sh -c 'tail -f "   21 hours ago        Up 21 hours         0.0.0.0:32922->22/tcp, 0.0.0.0:32921->4401/tcp, 0.0.0.0:32920->4403/tcp, 0.0.0.0:32919->4411/tcp, 0.0.0.0:32918->8080/tcp   workspaceuejf119p8lml5zt4_machinerjvjqwat0s9rid3r_che_dev-machine
cebff1ec00c8        golang:1.7                                                                      "bash"                   40 hours ago        Up 40 hours                                                                                                                                     sad_hoover
511af1e004f9        eclipse/che-server:5.4.1                                                        "/entrypoint.sh"         2 days ago          Up 2 days           8000/tcp, 0.0.0.0:8080->8080/tcp

$ docker inspect -f '{{.NetworkSettings.IPAddress}}' 009
172.17.0.5

$ ssh user@`docker inspect -f '{{.NetworkSettings.IPAddress}}' 009`
Read from socket failed: Connection reset by peer
```

I modified the Dockerfile as you suggested before trying to SSH into above container. But that didn't help. So I'm trying to understand how I should check if SSH works OK.

Comment (dharmit) [inline recipes/centos_wildfly_swarm/Dockerfile]: @eivantsov ping

File: recipes/centos_wildfly_swarm/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+FROM registry.centos.org/centos/centos
+
+MAINTAINER Dharmit Shah
+
+ARG JAVA_VERSION=1.8.0
+ARG SWARM_VERSION=2017.3.2
```
Comment (kenfinnigan) [inline recipes/centos_wildfly_swarm/Dockerfile]: @dharmit Not super familiar with stuff. Does this version need to be updated on every WF Swarm release?

Comment (dharmit) [inline recipes/centos_wildfly_swarm/Dockerfile]: @kenfinnigan if you'd like the WildFly Swarm stack to stay updated, yes.

Comment (kenfinnigan) [inline recipes/centos_wildfly_swarm/Dockerfile]: ok thanks

**Comment by eivantsov** [comment]
Image pushed - https://hub.docker.com/r/eclipse/centos_wildfly_swarm/

**Pull request merged**